ssl/t1_lib.c

```diff
@@ -2780,16 +2775,23 @@ int ssl_parse_serverhello_tlsext(SSL *s, PACKET *pkt)
     return 1;
 }
 
+static RAW_EXTENSION *get_extension_by_type(RAW_EXTENSION *exts, size_t numexts,
+                                            unsigned int type)
+{
+    size_t loop;
+
+    for (loop = 0; loop < numexts; loop++) {
```

Comment:
And I think the compiler is actually smart enough to turn that into a pointer itself. And it might also do some other optimizations depending on in which direction you go. But I really don't see the point.

I think what @richsalz really wanted to suggest was:

    for (ext = exts + numexts; ext != exts; ext--)
        if (ext->type == type)
            return ext;